bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/SearchResultServlet.java

```diff
@@ -16,282 +16,291 @@
-        if (resultsSize != 0) {
-            query.setHitsPerPage(resultsSize);
+        if (searchComponent.getResultsSize() != 0) {
+            query.setHitsPerPage(searchComponent.getResultsSize());
         }
         if (resultsOffset != 0) {
             query.setStart(resultsOffset);
         }
         SearchResult searchResult = query.getResult();
 
-        List<Hit> hits = searchResult.getHits();
-        if (hits != null) {
-            for (Hit hit : hits) {
-                try {
-                    Resource hitRes = hit.getResource();
-                    Page page = getPage(hitRes);
-                    if (page != null) {
-                        results.add(new PageListItemImpl(request, page, getId(searchResource),
-                                PageListItemImpl.PROP_DISABLE_SHADOWING_DEFAULT, null));
-                    }
-                } catch (RepositoryException e) {
-                    LOGGER.error("Unable to retrieve search results for query.", e);
+        // Query builder has a leaking resource resolver, so the following work around is required.
+        ResourceResolver leakingResourceResolver = null;
+        try {
+            Iterator<Resource> resourceIterator = searchResult.getResources();
+            while (resourceIterator.hasNext()) {
+                Resource resource = resourceIterator.next();
+
+                // Get a reference to QB's leaking resource resolver
+                if (leakingResourceResolver == null) {
+                    leakingResourceResolver = resource.getResourceResolver();
                 }
+
+                Optional.of(resource)
+                    .map(res -> resourceResolver.getResource(res.getPath()))
+                    .map(pageManager::getContainingPage)
+                    .map(page -> new PageListItemImpl(request, page, searchComponent.getId(),
+                        PageListItemImpl.PROP_DISABLE_SHADOWING_DEFAULT, null))
+                    .ifPresent(results::add);
+            }
+        } finally {
+            if (leakingResourceResolver != null) {
+                leakingResourceResolver.close();
             }
         }
         return results;
     }
 
-    private String getId(Resource resource) {
-        if (resource == null) {
-            return null;
+    /**
+     * A fall-back implementation of the Search model.
+     */
+    private final class DefaultSearch implements Search {
+
+        /**
+         * The search root page path.
+         */
+        @NotNull
+        private final String searchRootPagePath;
+
+        /**
+         * Construct the default search.
+         *
+         * @param currentPage The current page.
+         * @param resourceResolver The resource resolver.
+         */
+        public DefaultSearch(@NotNull final Page currentPage, @NotNull final ResourceResolver resourceResolver) {
+            this.searchRootPagePath = Optional.ofNullable(currentPage.getContentResource())
```

Comment:
I'm not convinced that the search root path should be set to the language root when a search root is not defined. Instead I would set it to `null`.

Reply:
I agree; however, please refer to the last section of the PR description. Specifically point #4.

This PR has been stale for a little but I believe the intention of this was to maintain the current behavior. of using the language root when the search root is not defined. It's a behavior that I don't like, but it already does this.